toolkit/authentication/src/androidTest/java/com/arcgismaps/toolkit/authentication/OAuthDefaultConfigurationTests.kt

```diff
@@ -0,0 +1,162 @@
+     * and fakes the token validation response.
+     *
+     * @since 200.5.0
+     */
+    @Test
+    fun successfulSignIn() = runTest {
+        val response = testOAuthChallengeWithStateRestoration {
+            InstrumentationRegistry.getInstrumentation().context.startActivity(getSuccessfulRedirectIntent("kotlin-authentication-test-1://auth"))
+        }.await().getOrNull()
+        assert(response is ArcGISAuthenticationChallengeResponse.ContinueWithCredential)
+        assert((response as ArcGISAuthenticationChallengeResponse.ContinueWithCredential).credential is OAuthUserCredential)
+    }
+
+    /**
+     * Given an [AuthenticatorState] configured with an [OAuthUserConfiguration] for ArcGIS Online,
+     * When an [ArcGISAuthenticationChallenge] is received and the user cancels the sign in process,
+     * Then the [ArcGISAuthenticationChallengeResponse] should be [ArcGISAuthenticationChallengeResponse.Cancel].
+     *
+     * @since 200.5.0
+     */
+    @Test
+    fun cancelSignIn() = runTest {
+        val response = testOAuthChallengeWithStateRestoration {
+            clickByText("Cancel")
+        }.await().exceptionOrNull()
+        assert(response is OperationCancelledException)
+    }
+
+    /**
+     * Given an [AuthenticatorState] configured with an [OAuthUserConfiguration] for ArcGIS Online,
+     * When an [ArcGISAuthenticationChallenge] is received and the user presses the back button,
+     * Then the [ArcGISAuthenticationChallengeResponse] should be [ArcGISAuthenticationChallengeResponse.Cancel].
+     *
+     * @since 200.5.0
+     */
+    @Test
+    fun pressBack() = runTest {
+        val response = testOAuthChallengeWithStateRestoration {
+            pressBack()
+        }.await().exceptionOrNull()
+        assert(response is OperationCancelledException)
+    }
+
+    /**
+     * Places an [Authenticator] in the composition and issues an OAuth challenge.
+     * Once the browser is launched, [userInputOnDialog] will be called to simulate user input.
+     * Also, the device will be rotated to ensure that the Authenticator can handle configuration changes
+     * before calling [userInputOnDialog].
+     *
+     * @since 200.5.0
+     */
+    @OptIn(ExperimentalCoroutinesApi::class)
+    fun TestScope.testOAuthChallengeWithStateRestoration(
+        userInputOnDialog: UiDevice.() -> Unit,
+    ): Deferred<Result<ArcGISAuthenticationChallengeResponse>> {
+        ArcGISEnvironment.configureArcGISHttpClient {
+            interceptTokenRequests()
+        }
+        val authenticatorState = AuthenticatorState().apply {
+            oAuthUserConfiguration = OAuthUserConfiguration(
+                "https://arcgis.com",
+                // This client ID is for demo purposes only. For use of the Authenticator in your own app,
+                //            // create your own client ID. For more info see: https://developers.arcgis.com/documentation/mapping-apis-and-services/security/tutorials/register-your-application/
```

Comment:
Comment needs formating
```suggestion
                // create your own client ID. For more info see: https://developers.arcgis.com/documentation/mapping-apis-and-services/security/tutorials/register-your-application/
```